lib/imperative-test.js

```diff
@@ -649,7 +649,14 @@ class ImperativeTest extends Test {
           else this.end();
         },
         e => {
-          this.fail('Promise rejection', e);
+          const lastResult = this.results[this.results.length - 1];
+          if (
+            !lastResult ||
+            lastResult.type !== 'fail' ||
+            lastResult.actual !== e
+          ) {
+            this.fail('Promise rejection', e);
```

Comment:
What about the following case:
```js
const err = new Error('message');
test.fail('with my error', err);
throw err;
```
This has nothing to do with `resolves()/rejects()` methods and I think the error shouldn't be ignored in that case.

Reply:
What if we added a custom error types `resolved`, `rejected`?